db/change_listener.go

```diff
@@ -36,16 +36,20 @@ type changeListener struct {
 	keyCounts             map[string]uint64      // Latest count at which each doc key was updated
 	OnDocChanged          DocChangedFunc         // Called when change arrives on feed
 	terminator            chan bool              // Signal to cause cbdatasource bucketdatasource.Close() to be called, which removes dcp receiver
+	checkpointPrefix      string                 // DCP checkpoint key prefix
+	sgCfgPrefix           string                 // SG config key prefix
 }
 
 type DocChangedFunc func(event sgbucket.FeedEvent)
 
-func (listener *changeListener) Init(name string) {
+func (listener *changeListener) Init(name string, groupID string) {
 	listener.bucketName = name
 	listener.counter = 1
 	listener.terminateCheckCounter = 0
 	listener.keyCounts = map[string]uint64{}
 	listener.tapNotifier = sync.NewCond(&sync.Mutex{})
+	listener.checkpointPrefix = base.DCPCheckpointPrefixWithGroupID(groupID)
```

Comment:
changeListener doesn't do any checkpointing, so this shouldn't actually be necessary to set (here or in the FeedArgs for changeListener).  I think I'd prefer to take it out, to reduce complexity and avoid giving the impression that changeListener is checkpointing.